libs/heattracesidesheet/src/lib/utils-sidesheet/useGetEleNetwork.ts

```diff
@@ -0,0 +1,43 @@
+import { ElectricalNetwork, useContextId, useHttpClient } from '@cc-components/shared';
+import { useQuery } from '@tanstack/react-query';
+import { HeatTrace } from '@cc-components/heattraceshared';
+
+export const useGetEleNetwork = (ht: HeatTrace | null) => {
+  const client = useHttpClient();
+  const contextId = useContextId();
+
+  if (!ht) {
+    throw new Error(`Failed to resolve the heat trace`);

```

Comment:
Failed to resolve what heat-trace? You shouldnt throw if heat-trace is null you should pause the query and resume it when the heat-trace is resolved..